Error on Linux while init'ing HCI

Here's a portion of the stacktrace:
```
panic: makechan: size out of range

goroutine 1 [running]:
panic(0x53e380, 0xc42000b1d0)
	/usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
github.com/currantlabs/ble/linux/hci.NewPool(0x5, 0xffffffffffffffff, 0xc420039de8)
	/Users/syoder/src/golang/src/github.com/currantlabs/ble/linux/hci/buffer.go:19 +0x3e
github.com/currantlabs/ble/linux/hci.(*HCI).Init(0xc420096000, 0x0, 0x0)
	/Users/syoder/src/golang/src/github.com/currantlabs/ble/linux/hci/hci.go:152 +0x754
github.com/currantlabs/ble/linux.NewDevice(0x7fc022118080, 0xc420018db0, 0x56a0b2)
	/Users/syoder/src/golang/src/github.com/currantlabs/ble/linux/device.go:21 +0xac
...
```

Looks like it's trying to create a pool of buffers here: https://github.com/currantlabs/ble/blob/master/linux/hci/hci.go#L151

But `bufCnt` is zero, so it's trying to create a pool with -1 buffers, which crashes.

And here's the twist: it only crashes when I'm NOT running `hcidump`.  When I am running it, `bufCnt` has a proper value (15).  Maybe there's a race condition, and `hcidump` changes the timing?

If we can get to the bottom of this, I have a couple of PR's I'd like to submit that will add the ability to advertise arbitrary manufacturer and service advertisements (to allow altBeacon and Eddystone advertisements) on both mac and linux.

--- a/linux/hci/socket/socket.go
+++ b/linux/hci/socket/socket.go
@@ -104,6 +104,15 @@ func open(fd, id int) (*Socket, error) {
 	if err := unix.Bind(fd, &sa); err != nil {
 		return nil, errors.Wrap(err, "can't bind socket to hci user channel")
 	}
+
+	// poll for 20ms to see if any data becomes available, then clear it
+	pfds := []unix.PollFd{unix.PollFd{Fd: int32(fd), Events: unix.POLLIN}}
+	unix.Poll(pfds, 20)
+	if pfds[0].Revents&unix.POLLIN > 0 {
+		b := make([]byte, 100)
+		unix.Read(fd, b)
+	}
+
 	return &Socket{fd: fd}, nil
 }
 

